Elevel/Application/Features/ApplicationUserFeatures/GetAllApplicationUserQuery.cs

```diff
@@ -0,0 +1,150 @@
+                    && string.IsNullOrWhiteSpace(request.FirstName) 
+                    && !string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                        x.LastName == request.LastName
+                        && x.Email == request.Email);
+                }
+                else 
+                if (string.IsNullOrWhiteSpace(request.LastName) 
+                    && !string.IsNullOrWhiteSpace(request.FirstName) 
+                    && !string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                        x.FirstName == request.FirstName
+                        && x.Email == request.Email);
+                }
+                else 
+                if (!string.IsNullOrWhiteSpace(request.LastName) 
+                    && string.IsNullOrWhiteSpace(request.FirstName) 
+                    && string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                        x.LastName == request.LastName);
+                }
+                else
+                if (string.IsNullOrWhiteSpace(request.LastName) 
+                    && string.IsNullOrWhiteSpace(request.FirstName) 
+                    && !string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                         x.Email == request.Email);
+                }
+                else 
+                if (string.IsNullOrWhiteSpace(request.LastName) 
+                    && !string.IsNullOrWhiteSpace(request.FirstName) 
+                    && string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                        x.FirstName == request.FirstName);
+                }
+                else 
+                if (string.IsNullOrWhiteSpace(request.LastName) 
+                    && string.IsNullOrWhiteSpace(request.FirstName) 
+                    && string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking();
+                }
+                else
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                        x.LastName == request.LastName
+                        && x.FirstName == request.FirstName
+                        && x.Email == request.Email);
+                }
+
+                return new Response()
+                {
+                    CurrentPage = request.CurrentPage,
+                    PageSize = request.PageSize,
+                    RowCount = users.Count(),
+                    Results = users.Skip(request.SkipCount()).Take(request.PageSize).ProjectTo<UserDto>(_mapper.ConfigurationProvider).ToList()
```

Comment:
it's better to use PagedResult, but I am OK with this, just use Async version where possible, so CountAsync and ToListAsync